custom_components/tahoma/cover.py

```diff
@@ -292,10 +292,27 @@ async def async_cancel_or_stop_cover(self, cancel_commands, stop_commands) -> No
 
         # Cancelling a running execution will stop the cover movement
         if exec_id:
-            await self.async_cancel_command(exec_id)
-        # Fallback to available stop commands when execution was initiated outside Home Assistant
-        else:
-            await self.async_execute_command(self.select_command(*stop_commands))
+            return await self.async_cancel_command(exec_id)
+
+        # Retrieve running executions from server for executions initiated outside Home Assistant
+        executions = await self.coordinator.client.get_current_executions()
+        exec_id = next(
+            (
+                execution.id
+                for execution in executions
+                for action in reversed(execution.action_group.get("actions"))
+                for command in action.get("commands")
+                if action.get("deviceurl") == self.device.deviceurl
+                and command.get("name") in cancel_commands
+            ),
+            None,
+        )
+
+        if exec_id:
+            return await self.async_cancel_command(exec_id)
+
+        # Fallback to available stop commands when no execution can be found
+        await self.async_execute_command(self.select_command(*stop_commands))
```

Comment:
Finally, why not keep only this one? Do you think some device does not have it?

Reply:
You mean, removing the other logic in favour of this one? Unfortunately the stop command don't work for every device. Even when a device has a stop command (like my pergola), it just doesn't work.

Cancelling a running execution is the method to stop a cover movement (also on tahomalink.com) and it is a reliable way that works with every device. This is some Somfy logic that I couldn't work around unfortunately and it was causing quite some user issues.

Reply:
@tetienne good point by the way. I updated the documentation to make this design choice more clear.